torch_geometric/nn/conv/cugraph/base.py

```diff
@@ -48,40 +50,48 @@ def to_csc(
             edge_attr (torch.Tensor, optional): The edge features.
                 (default: :obj:`None`)
         """
+        if size is None:
+            warnings.warn(
+                "Inferring graph size from 'edge_index' causes a "
+                "decline in performance and does not work for "
+                "bipartite graphs. To suppress this warning, pass in "
+                f"'size' explicitly in {__name__}.to_csc().")
+            max_id = edge_index.max().item()
+            num_src_nodes = num_dst_nodes = max_id + 1
+        else:
+            num_src_nodes, num_dst_nodes = size
+
         row, col = edge_index
-        num_target_nodes = size[1] if size is not None else int(col.max() + 1)
-        col, perm = index_sort(col, max_value=num_target_nodes)
+        col, perm = index_sort(col, max_value=num_dst_nodes)
         row = row[perm]
 
-        colptr = index2ptr(col, num_target_nodes)
+        colptr = index2ptr(col, num_dst_nodes)
 
         if edge_attr is not None:
-            return (row, colptr), edge_attr[perm]
+            return (row, colptr, num_src_nodes), edge_attr[perm]
 
-        return row, colptr
+        return row, colptr, num_src_nodes
 
     def get_cugraph(
         self,
-        num_src_nodes: int,
-        csc: Tuple[Tensor, Tensor],
+        csc: Tuple[Tensor, Tensor, int],
         max_num_neighbors: Optional[int] = None,
     ) -> Any:
         r"""Constructs a :obj:`cugraph` graph object from CSC representation.
         Supports both bipartite and non-bipartite graphs.
 
         Args:
-            num_src_nodes (int): The number of source nodes.
-            csc ((torch.Tensor, torch.Tensor)): A tuple containing the CSC
+            csc ((torch.Tensor, torch.Tensor, int)): A tuple containing the CSC
                 representation of a graph, given as a tuple of
-                :obj:`(row, colptr)`. Use the :meth:`CuGraphModule.to_csc`
-                method to convert an :obj:`edge_index` representation to the
-                desired format.
+                :obj:`(row, colptr, num_src_nodes)`. Use the
+                :meth:`CuGraphModule.to_csc` method to convert an
+                :obj:`edge_index` representation to the desired format.
             max_num_neighbors (int, optional): The maximum number of neighbors
                 of a target node. It is only effective when operating in a
                 bipartite graph. When not given, will be computed on-the-fly,
                 leading to slightly worse performance. (default: :obj:`None`)
         """
-        row, colptr = csc
+        row, colptr, num_src_nodes = csc
```

Comment:
it would be good to add a comment in this function mentioning that since in both MFG and FG cases, we end up creating the structure based on this `num_src_nodes`, we will check the size of the inputs for an operator in the bindings correctly (we actually check the size of the input against `num_src_nodes` which was given independent of the size of the input here).

Reply:
I guess if we do the check for input feature size within the operators (before calling into this function), we should simply have a comment here saying that the size was already checked instead of what I mentioned above.
If we move that check to the bindings though, my comment above holds